desk/app/expose.hoon

```diff
@@ -261,56 +328,107 @@
+        =/  kind=path
           ?-  -.nest
-            %chat   [%chat ~]
-            %diary  [%diary '' '']
-            %heap   [%heap ~]
+            %chat   /chat
+            %diary  /diary
+            %heap   /heap
           ==
         =/  ref=cite:c
-          (from-post:cite:u nest id.r-channel kind-data)
+          (from-post:cite:u nest id.r-channel kind)
         ?.  (~(has in open) ref)  [~ this]
         =.  open  (~(del in open) ref)
         :_  this
-        [(clear-page:e ref) (refresh-widget:e bowl open)]
+        :-  (clear-page:e ref)
+        =-  (murn - same)
+        :~  (refresh-widget:e bowl open)
+            (inflate-contacts-profile:e [our now]:bowl open)
+        ==
       ==
     ==
+  ::
+      [%contacts %prime *]
+    ::  we don't actually care what happens after we try to prime the cache,
+    ::  we just no-op and assume channels agent will kick our subscription
+    ::  once we hear back.
+    ::
+    [~ this]
   ::
       [%contacts %news ~]
     ?-  -.sign
       %poke-ack  !!
-      %kick      [[%pass /contacts/news %agent [our.bowl %contacts] %watch /news]~ this]
+      %kick      [[%pass /contacts/news %agent [our.bowl %contacts] %watch /v1/news]~ this]
     ::
         %watch-ack
       ?~  p.sign  [~ this]
       ~&  >>>  [dap.bowl %rejected-by-contacts]
       [~ this]
     ::
         %fact
-      ::  our own contact details changes. assume this affects all pages we're
-      ::  serving, refresh the cache.
-      ::  note that we don't do this kind of reactivity for contact details of
-      ::  other ships. at the extreme, reacting to that accurately in a non-
-      ::  wasteful way would require trawling all content for relevane, which
-      ::  in turn is slow. if we get to the point of wanting that to be
-      ::  fresh(er), we should just set an hourly timer that re-render the
-      ::  entire cache.
+      =+  !<(=response:co q.cage.sign)
+      ?+  -.response  [~ this]
+          %self
+        ::  our own contact details changed. assume this affects all pages
+        ::  we're serving, refresh the cache.
+        ::  note that we don't do this kind of reactivity for contact details
+        ::  of other ships. at the extreme, reacting to that accurately in a
+        ::  non-wasteful way would require trawling all content for relevance,
+        ::  which in turn is slow. if we get to the point of wanting that to be
+        ::  fresh(er), we should just set an hourly timer that re-render the
+        ::  entire cache.
+        ::
+        :_  this
+        %+  weld
+          (drop (refresh-widget:e bowl open))
+        (refresh-pages:e bowl ~(tap in open))
       ::
-      =+  !<(=news-0:co q.cage.sign)
-      ?.  =(our.bowl who.news-0)  `this
-      :_  this
-      %+  weld
-        (refresh-widget:e bowl open)
-      (refresh-pages:e bowl ~(tap in open))
+          %peer
+        ::  someone else's contact details changed. if they have any pinned
```

Comment:
I worry about the performance here, this means anyone that pins something will have to answer N subscriptions times the number of peers. For people that have 5000+ contacts from the old days, this could immediately DDOS the ship no?

Reply:
At the very least, we should do the `TODO` from below, where we check for a cache entry locally, and don't go over the network if we already have one.